csrc/aes_kwp.cpp

```diff
@@ -0,0 +1,96 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+#include <cstdio>
+#include <cassert>
+#include <algorithm> // for std::min
+#include <openssl/aes.h>
+#include <openssl/evp.h>
+#include <openssl/err.h>
+#include "generated-headers.h"
+#include "util.h"
+#include "env.h"
+#include "buffer.h"
+#include "keyutils.h"
+
+using namespace AmazonCorrettoCryptoProvider;
+
+JNIEXPORT int JNICALL Java_com_amazon_corretto_crypto_provider_AesKeyWrapSpi_wrapKey(
+  JNIEnv *pEnv,
+  jclass,
+  jbyteArray keyArray,
+  jbyteArray inputArray,
+  jbyteArray outputArray
+)
+{
+    try {
+        raii_env env(pEnv);
+
+        java_buffer key = java_buffer::from_array(env, keyArray);
+        java_buffer input = java_buffer::from_array(env, inputArray);
+        java_buffer output = java_buffer::from_array(env, outputArray);
+
+        AES_KEY aes_key;
+        SecureBuffer<uint8_t, 32> keybuf;
+        key.get_bytes(env, keybuf.buf, 0, key.len());
+        if (AES_set_encrypt_key(keybuf.buf, key.len()*8, &aes_key) != 0) {
+            throw java_ex::from_openssl(EX_RUNTIME_CRYPTO, "AES key init failed");
```

Comment:
Generally we do the following instead (which is equivalent). It's up to you:
```suggestion
            throw_openssl(EX_RUNTIME_CRYPTO, "AES key init failed");
```